### DISCUSSION TITLE
Can I plot data on a chart but disable interactivity for certain points?

### DISCUSSION THREAD

**Discussion by pulsedemon** [recharts/recharts#4704] [Q&A]
Hi,
I have a line chart with a lot of data points, and I want to enable dots for specific points and only have those points be interactive.  So if you hover in-between those points, the tooltip should snap to the nearest point with a dot.  So far I haven't found an easy way to do this so I wanted to ask if anyone knows if this is possible.  If you need an example, let me know.
Thank you

**Comment by ckifer** [comment]
Hi! It's definitely possible to render dots for only certain points, but I don't think it's possible to only make those ones interactive. 

Answering on mobile at the moment but if you start a codesandbox I can help you out in a bit with the first part at least.

**Reply by pulsedemon** [reply]
hey Coltin, thanks for the quick reply! I think I already have the first part figured out.  See this codesandbox: https://codesandbox.io/p/sandbox/line-chart-width-xaxis-padding-forked-v4z99s?file=%2Fsrc%2FApp.tsx%3A80%2C1
The dot for the second point "10/11/10" isn't rendering as expected, but I can still hover on it and the tooltip shows.  I'm going to try and figure out a way to handle this but if you think of anything please let me know!

For context, I have some charts that display monthly and quarterly data and I only want the user to be able to see the monthly/quarterly tooltips, but still show the daily movement between those points.

**Reply by ckifer** [reply]
Cool - I'll try to help when I can. I have a few ideas but nothing I'm sure about.

**Reply by pulsedemon** [reply]
thanks.  if it's not possible to do with the current api, i could look into opening a pr.  didn't wanna go that route without discussing options first though

**Reply by ckifer** [reply]
Just a note that we are currently working on 3.x and contributions will _hopefully_ be much easier after the release of 3.0. The current code is... hard to follow to say the least

**Reply by pulsedemon** [reply]
lol ok good to know.  is there a timeline for the 3.x release?  I'm in the middle of moving my company's charts from chartjs to recharts so the sooner the better

**Reply by ckifer** [reply]
Unfortunately we're just 2 volunteer maintainers at the moment. We're trying, but spare time is spare time and it can't really be estimated. We're trying for the next few months, but that is subject to change.

Try this one once, I added a custom tooltip which now shouldn't show any tooltips when the certain datapoints are hovered. The snapping behavior is going to be the difficult part.

[![Edit line-chart-width-xaxis-padding (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/line-chart-width-xaxis-padding-forked-cvrtrg)

**Reply by pulsedemon** [reply]
gotcha.  If you need any help, maybe I can contribute some time.  lmk
I don't see your changes, but I was also able to get that far in some previous experiments by creating a custom tooltip and returning null or something based on the showTooltip attribute.  Yeah, seems like the snapping is the main thing to figure out.

**Reply by ckifer** [reply]
Yeah we absolutely need contributors. Feel free to join our slack channel [here](https://join.slack.com/t/recharts/shared_invite/zt-1o9t7luva-x9CbZ1_nI2OHXMr6id1T4A).

Oops fixed that embed link now, but yeah same deal

**Reply by ckifer** [reply]
Also `activeDot` can be used to control the dot on hover, same as `dot` - that just needs a bit of tweaking `CustomDot` to work correctly.

`cursor` on `Tooltip` can be used to control the cursor line, but it unfortunately doesn't support a function w/props at the moment.

The "snapping" functionality you're really looking for is all internal to recharts which is why I mentioned I don't think its currently possible to do.

**Reply by n8urm8** [reply]
I'm working with pulsedemon on this issue, and I'm wondering, is there a way to get a list of all x-coordinates and the related label, or just a list of x-coordinates that are used for the data points? Or is there maybe a way to extrapolate this based on the width of the chart area and the domain of the x-axis?

Edit: I'm currently using the custom cursor component to determine the nearest index of data that should be shown, then using the width prop divided by the length of the data (since we use that to determine the xAxis) plus the left prop to determine the x pixel position of the nearest data point to be shown. It seems to work, but it's slightly off the actual cursor point by a pixel or 2 in some cases. I don't think the tooltip and active dot components have access to all the needed props, so I'll probably just store this x point and index to use for those components. This seems to create the 'snapping' effect we want.

**Reply by ckifer** [reply]
@n8urm8 Not recommended as it will break in 3.0, but you can use `Customized` to get all internal chart state.

See https://recharts.org/en-US/examples/CustomizedRectangleContent for an example 

See https://github.com/recharts/recharts/discussions/4437 for why it will break in 3.0

You might also just be able to get this all from using a custom label component for `LabelList`

**Reply by n8urm8** [reply]
Thanks, I'll look into that! Hah, guess I'm not the only one trying to do this

> ...The most important thing (imo) is to be able to translate two values X and Y (i.e. "Page A", 2000 or 2000, 2000) into chart coordinates. That doesn't even exist in Customized today and would be incredibly powerful

**Reply by ckifer** [reply]
right. We'll get there!
